python/lsst/dax/ppdbx/gcp/bq.py

```diff
@@ -33,6 +39,25 @@
 from .env import require_env
 
 
+def check_dataset_exists(project_id: str, dataset_id: str) -> None:
+    """Check if a BigQuery dataset exists.
+
+    Parameters
+    ----------
+    project_id : `str`
+        Google Cloud project ID.
+    dataset_id : `str`
+        Fully qualified BigQuery dataset ID including project name.
+
+    Raises
+    ------
+    google.api_core.exceptions.NotFound
+        If the specified dataset does not exist.
```

Comment:
```suggestion
        Raised if the specified dataset does not exist.
```
It may be better to catch google exception and raise `LookupError` so that clients could catch a specific exception type instead of generic `Exception`. Or even return a boolean instead.

Reply:
I like your idea of re-raising as `LookupError` instead of the Google type - I've changed that throughout.